apps/patient/src/utils/general.ts

```diff
@@ -207,3 +211,40 @@ export const convertReadyByToUTCTimestamp = (readyBy: string, readyByDay: string
 };
 
 export const capitalize = (str: string) => str.charAt(0).toUpperCase() + str.slice(1);
+
+/**
+ * Calculates the total price of all fills in an order based on the pricing information
+ * available in a Walmart medication list. All fills must match the pricing information
+ * to be included in the total price calculation.
+ *
+ * @param {types.Fill[]} fills - An array of fills representing the medications in the order.
+ * @returns {number} The total price of the order.
+ */
+export const getTotalWalmartOrderPrice = (fills: types.Fill[]): number => {
+  let totalPrice = 0;
+  let allFillsMatch = true;
+
+  for (const fill of fills) {
+    // Check if the fill is in the walmartMedLookup list
+    const match = walmartMedLookup.find(
+      (med) =>
+        new RegExp(`${med.name}.*${med.strength}`, 'i').test(fill.treatment.name) &&
```

Comment:
Can we not just do an exact match here? Theoretically worried about partial matches. For example

`Acetaminophen 500mg - Caffeine 60mg - Pyrilamine maleate 15mg` (excedrin)

would match `Acetaminophen 15mg` despite not being the same thing. (obviously a contrived example)

Reply:
I think that might be hard to ensure an exact match and the example might be an edge case, IMO this should be good to go because this is at max a 2 or 3 week test that will affect 200 - 400 orders. I think getting this out today is more the higher priority since we have an upcoming meeting with Walmart, hoping to show some positive results